src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -456,12 +455,7 @@ public async Task DispatchEvent(string eventDescriptorJson, string eventArgsJson
             {
                 // JsonSerializerOptions are tightly bound to the JsonContext. Cache it on first use using a copy
                 // of the serializer settings.
-                if (_jsonContext is null)
-                {
-                    _jsonContext = new(new JsonSerializerOptions(JSRuntime.ReadJsonSerializerOptions()));
-                }
-
-                webEventData = WebEventData.Parse(Renderer, _jsonContext, eventDescriptorJson, eventArgsJson);
+                webEventData = WebEventData.Parse(Renderer, JSRuntime.ReadJsonSerializerOptions(), eventDescriptorJson, eventArgsJson);
```

Comment:
Is there a way we could push this logic all the way down to the serializer level? (Keep the logic within WebEventData and avoid the intermediate JsonElement deserialization)

Or alternatively, defer any deserialization until we are out of the serializer.

Reply:
Let's do this in a follow up. I want to unblock the size increase issues for WASM for preview7